docs/contributing/contributing.rst

```diff
@@ -397,6 +404,66 @@ Pull Requests
   .. _Co-authored-by: https://github.blog/2018-01-29-commit-together-with-co-authors/
   .. _Issue #2816: https://github.com/DataBiosphere/toil/issues/2816
   .. _toil.lib.retry: https://github.com/DataBiosphere/toil/blob/master/src/toil/lib/retry.py
+  
+
+Publishing a Release
+~~~~~~~~~~~~~~~~~~~~
+
+These are the steps to take to publish a Toil release:
+
+* Determine the release version **X.Y.Z**. This should follow
+  `semantic versioning`_; if user-workflow-breaking changes are made, **X**
+  should be incremented, and **Y** and **Z** should be zero. If non-breaking
+  changes are made but new functionality is added, **X** should remain the same
+  as the last release, **Y** should be incremented, and **Z** should be zero.
+  If only bugfixes are released, **X** and **Y** should be the same as the last
+  release and **Z** should be incremented.
```

Comment:
In practice, I believe we've only incremented **Z** for hot fixes.  I know **Z** for bug fixes only (another source said each commit increments **Z** by one) seems to be the more common convention, but versioning conventions also seem to be custom-appropriated by project from what I've found, for example:

 - Ubuntu versions are "Year.Month". For example, 10.04 was released in April 2010.
 - TeX versions are theoretically only bux-fixes forevermore, so their versions are asymptotically approaching pi (e.g. 3.1415926)

So I'd argue for keeping it for hot-fixes only (though this seems to rarely happen, so maybe we're wasting a perfectly good number slot).

Reply:
What's the difference between a hotfix and a bugfix?

Reply:
I've changed it to say "patches", which maybe means the same thing as hotfixes?

Reply:
I suppose a hotfix is, in my mind, when a release is done, and we realize that Toil, say, no longer reads credentials or something and we immediately have to patch it in order to get it working again.  I suppose there's a grey area and patch seems like a better term, thank you!